source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -331,14 +333,33 @@ void Filter::closeStream() {
   }
 }
 
+void Filter::deferredCloseStream() {
+  ENVOY_LOG(debug, "Calling deferred close on stream");
+  config_->threadLocalStreamManager().deferredErase(this, filter_callbacks_->dispatcher());
```

Comment:
Why do you need to keep the filter object around?

Reply:
Sorry, I might not fully get the question. 

 Do you mean `this` pointer? It is just the address that serve as ID/key of the map --- `absl::flat_hash_map<Filter*, ExternalProcessorStreamPtr> stream_manager_`.   I don't keep the filter object around for other purpose.

This map is shared with multiple filters per thread. And key/this pointer here is used for deletion.

Reply:
I see. This can lead to duplicate keys in the stream_manager_ map, since the filter object can be deleted and a new one allocated at the exactly same address.

Reply:
Yeah, perhaps we should have an instance UUID instead.

Reply:
Good point! I have switched to [stream_id](https://github.com/envoyproxy/envoy/blob/bb4a76ac13d0b8a6851e7db4b1c3a4f2d408f92e/source/common/http/conn_manager_impl.cc#L781) for single request/response (i.e., unique id for ext_proc filter)